pkg/targetallocator/deployment.go

```diff
@@ -52,6 +52,7 @@ func Deployment(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTele
 					ServiceAccountName: ServiceAccountName(otelcol),
 					Containers:         []corev1.Container{Container(cfg, logger, otelcol)},
 					Volumes:            Volumes(cfg, otelcol),
+					ImagePullSecrets:   otelcol.Spec.ImagePullSecrets,
```

Comment:
I really have no idea how target allocators work yet, so please let me know if this should be reverted / changed

Reply:
Shouldn't the same field be mixed with collector? Instead use this field: `otelcol.Spec.TargetAllocator.ImagePullSecrets`